core/src/OpenACC/Kokkos_OpenACC_FunctorAdapter.hpp

```diff
@@ -19,16 +19,41 @@
 
 #include <type_traits>
 
+#define KOKKOS_OPENACC_CONTAIN_SEQLOOP 0
+#define KOKKOS_OPENACC_CONTAIN_WORKERLOOP 1
+
 namespace Kokkos::Experimental::Impl {
 
-template <class Functor, class Policy>
+template <class Functor, class Policy, int N, typename = void>
```

Comment:
There is no need for these` KOKKOS_OPENACC_CONTAIN_{SEQ,WORKER}LOOP` macros.
Please prefer a strong enumeration (or tag classes).  Something like
```C++
enum class RoutineClause { worker, seq };

template <class Functor, class Policy, RoutineClause>
class FunctorAdapter;

#define KOKKOS_IMPL_ACC_FUNCTOR_ADAPTER(CLAUSE)                    \
  template <class Functor, class Policy>                           \
  class FunctorAdapter<Functor, Policy, RoutineClause::CLAUSE> {   \
    Functor m_functor;                                             \
    using WorkTag = typename Policy::work_tag;                     \
                                                                   \
   public:                                                         \
    FunctorAdapter(Functor const &functor) : m_functor(functor) {} \
    KOKKOS_IMPL_ACC_PRAGMA_HELPER(routine CLAUSE)                  \
    template <class... Args>                                       \
    KOKKOS_FUNCTION void operator()(Args &&...args) const {        \
      if constexpr (std::is_void_v<WorkTag>) {                     \
        m_functor(static_cast<Args &&>(args)...);                  \
      } else {                                                     \
        m_functor(WorkTag(), static_cast<Args &&>(args)...);       \
      }                                                            \
    }                                                              \
  }

KOKKOS_IMPL_ACC_FUNCTOR_ADAPTER(worker);
KOKKOS_IMPL_ACC_FUNCTOR_ADAPTER(seq);

#undef KOKKOS_IMPL_ACC_FUNCTOR_ADAPTER

Reply:
Updated as suggested.